frame/evm/src/runner/stack.rs

```diff
@@ -136,14 +136,7 @@ where
 		//
 		// EIP-3607: https://eips.ethereum.org/EIPS/eip-3607
 		// Do not allow transactions for which `tx.sender` has any code deployed.
-		//
-		// We extend the principle of this EIP to also prevent `tx.sender` to be the address
-		// of a precompile. While mainnet Ethereum currently only has stateless precompiles,
-		// projects using Frontier can have stateful precompiles that can manage funds or
-		// which calls other contracts that expects this precompile address to be trustworthy.
-		if is_transactional
-			&& (!<AccountCodes<T>>::get(source).is_empty() || precompiles.is_precompile(source))
-		{
+		if is_transactional && !<AccountCodes<T>>::get(source).is_empty() {
```

Comment:
`is_precompile` now takes an extra argument being the remaining gas, and if not OOG will return the bool and the cost of performing that check. This function is used in only 2 places in frontier and evm:

- evm: inside is_cold, which occurs when we interact with a smart contract (getting code, doing a subcall)
- frontier (where this comment is): to know if the `from` address is the one of a precompile. EIP-3607 forbids `from` to be the address of a smart contract, and we **extended it** in frontier so that it cannot be the one of a precompile. The precompile part is NOT part of the EIP.

Now it is required to take into account the cost of calling `is_precompile`, which is less trivial than the change in the EVM. In this frontier code it is a non trivial change, I thus suggest we get rid of that part to keep things simple (we could maybe deduct the cost from the gas limit or whatever, but it may have unwanted side effects).